### PR TITLE
Improve Tests using tapejs

### PR DESCRIPTION
**Pull request by rajeshsegu**
I hated the tests in this library as they used just colors to know if a test is passed or failed. I have implemented tapejs tests for this module to make sure we can integrate them with any ci to validate every commit that goes into this module.

now you can run:

# npm run test

1..26
tests 26
pass  26
ok

# npm run coverage

----------------------|----------|----------|----------|----------|----------------|
File                  |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
----------------------|----------|----------|----------|----------|----------------|
 node-maxmind-db/     |    95.45 |      100 |    85.71 |    95.45 |                |
  index.js            |    95.45 |      100 |    85.71 |    95.45 |             39 |
 node-maxmind-db/lib/ |    87.42 |    74.36 |    80.33 |     88.4 |                |
  Decoder.js          |    86.78 |    71.01 |    85.71 |     88.3 |... 260,301,318 |
  IPParser.js         |      100 |      100 |      100 |      100 |                |
  Metadata.js         |    82.86 |      100 |       50 |    82.86 |... 28,32,36,40 |
  Reader.js           |    85.37 |    72.22 |    86.67 |    86.42 |... 132,134,154 |
----------------------|----------|----------|----------|----------|----------------|
All files             |    87.93 |    75.21 |    80.88 |    88.86 |                |
----------------------|----------|----------|----------|----------|----------------|

### CODE REVIEW

File: lib/IPParser.js [hunk 1]
```diff
@@ -1,44 +1,49 @@
...
-	if(l < 8) ipstr = ipstr.replace('::',repeat(':',10 - l));
-	ipstr.split(':').forEach(function(hex,i){
-		if(hex=="")return;
-		if(hex.length < 4){
-			hex = repeat('0',4 - hex.length) + hex;
-		}
-		arr.write(hex,i*2,'hex');
-	});
-	return arr;
+var repeat = function repeat(c, l) {
```
Comment (EaterOfCode) [inline lib/IPParser.js]: you're defining `repeat` twice, first as function then as variable. is this intended?

File: test/test-ipparser.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    verify(t, '2001:db8:85a3::8a2e:370:7334');
+    t.end();
+  });
+
+  it('should throw exception on invalid ipv4 formats', function should(t) {
+    verifyThrows(t, 'myipaddress');
+    verifyThrows(t, '::ffff:192.0.2.128');
+    t.end();
+  });
+});
```
Comment (EaterOfCode) [inline test/test-ipparser.js]: No need to test output?

File: lib/IPParser.js (hunk 1 above)
Comment (rajeshsegu) [inline lib/IPParser.js]: was a mistake. will fix it.

File: test/test-ipparser.js (hunk 2 above)
Comment (rajeshsegu) [inline test/test-ipparser.js]: I am going to submit another pull request, where I am introducing 'ip-address' module for better ip validation.

**Comment by rajeshsegu** [comment]
Also, is there a reason why there is 4 space indent at places and 2 space indent at other places ? I like the 2 space indent better. If we agree on something we can make this more uniform. Thanks.

**Comment by rajeshsegu** [comment]
@EaterOfCode @PaddeK any thing more that you guys want me to fix ?

File: lib/IPParser.js [hunk 3]
```diff
@@ -1,44 +1,59 @@
...
-	if(l < 8) ipstr = ipstr.replace('::',repeat(':',10 - l));
-	ipstr.split(':').forEach(function(hex,i){
-		if(hex=="")return;
-		if(hex.length < 4){
-			hex = repeat('0',4 - hex.length) + hex;
-		}
-		arr.write(hex,i*2,'hex');
-	});
-	return arr;
-};
```
Comment (EaterOfCode) [inline lib/IPParser.js]: does it make sense to make a wrapper for ip-address package instead of just directly using it?

**Comment by EaterOfCode** [comment]
sorry for the late reply, my personal favor is on 4 spaces as the original code was 4 space indent. I may have ruined it by mixing tabs with it. the code I wrote for this project is extremely bad. (Im sorry @PaddeK). the only thing that really still needs to be fixed is caching as lookups take alot of time.

I have made a patch for adding path lookup and ipv4 start node caching but never found the motivation afterwards to finish it: [here](https://github.com/PaddeK/node-maxmind-db/compare/master...EaterOfCode:moreSpeed).

File: lib/IPParser.js (hunk 3 above)
Comment (rajeshsegu) [inline lib/IPParser.js]: Done.

**Comment by rajeshsegu** [comment]
@EaterOfCode @PaddeK Please do review

+ added .editorconfig to the project to make sure that code feels more uniform

**Comment by rajeshsegu** [comment]
@EaterOfCode @PaddeK  lets merge this if you guys do not have more review comments. Thanks in advance.
